File: app/src/main/java/com/halo/redpacket/HomeActivity.java
package com.halo.redpacket;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import com.halo.redpacket.activity.PieImageActivity;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class HomeActivity extends AppCompatActivity {

    private Class[] arrays = new Class[] {
            PieImageActivity.class,
            MainActivity.class
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home);

        ListView listView = (ListView) findViewById(R.id.listView);
        List<Class> list = Arrays.asList(arrays);
        listView.setAdapter(new ArrayAdapter<Class>(this, android.R.layout.simple_list_item_1, list));
        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                Intent intent = new Intent(HomeActivity.this, arrays[position]);
                startActivity(intent);
            }
        });
    }
}

File: app/src/main/java/com/halo/redpacket/HomeActivity.kt
package com.halo.redpacket

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.AdapterView.OnItemClickListener
import android.widget.ArrayAdapter
import android.widget.ListView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.halo.redpacket.activity.PieImageActivity
import com.halo.redpacket.inject.*
import com.halo.redpacket.mvvm.LifeCycleListener
import java.util.*
import javax.inject.Inject

class HomeActivity : AppCompatActivity() {
    private val arrays = arrayOf<Class<*>>(
            PieImageActivity::class.java,
            MainActivity::class.java,
            PmActivity::class.java
    )

    @Inject
    lateinit var user: UserInject

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_home)
        val listView = findViewById<View>(R.id.listView) as ListView
        val list = Arrays.asList(*arrays)
        listView.adapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, list)
        listView.onItemClickListener = OnItemClickListener { parent, view, position, id ->
            val intent = Intent(this@HomeActivity, arrays[position])
            startActivity(intent)
        }

        val userModule3 = UserModule3()
        userModule3.param("userModule")
        DaggerUserComponent3.builder().userModule3(userModule3).build().inject(this)
        Toast.makeText(this, user.testInjectWithParam(), Toast.LENGTH_LONG).show()

        //绑定 LifeCycleListener 对象
        lifecycle.addObserver(LifeCycleListener())
    }
}